assets/js/script.js

```diff
@@ -0,0 +1,113 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+
+// declaring password criteria
+const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const lowercase = "abcdefghijklmnopqrstuvwxyz";
+const numeric = "0123456789";
+const specialCharacters = " !\"#$%&'()*+,-./:;<=>?@[]^_`{|}~";
+let userPasswordConfirms;
+
+// prompt user password length input and validate
+const getPasswordLength = () => {
+  const passwordLength = prompt(
+    "How many characters do you want your password to have. Please enter between 8 and 128 characters"
+  );
+  const passLengthNum = parseInt(passwordLength, 10);
+  //validate user password length input
+  if (passLengthNum >= 8 && passLengthNum <= 128) {
+    // return the password length
+    return passLengthNum;
+  } else {
+    // alert user to re-try input a correct number
+    alert("Please enter a valid number");
+    return;
+  }
+};
+
+//prompting user to select the password criteria's
+function passwordCriteria() {
+  const finalRandomPasswordArray = [];
+  // declaring variables for all criteria
+  const hasUppercase = confirm(
+    "Would you like to have uppercase characters in you password"
+  );
+  const hasLowercase = confirm(
+    "Would you like to have lowercase characters in you password"
+  );
+  const hasNumeric = confirm(
+    "Would you like to have numeric characters in you password"
+  );
+  const hasSpecialCharacter = confirm(
+    "Would you like to have uppercase characters in you password"
+  );
+  // pushing each criteria arrays the user selects to the final array
+  if (hasLowercase || hasUppercase || hasSpecialCharacter || hasNumeric) {
+    if (hasUppercase) {
+      // push if user has uppercase characters confirmed
+      finalRandomPasswordArray.push(uppercase);
+    }
+    // push if user has lowercase character confirmed
+    if (hasLowercase) {
+      finalRandomPasswordArray.push(lowercase);
+    }
+    // push if user has has numeric character confirmed
+    if (hasNumeric) {
+      finalRandomPasswordArray.push(numeric);
+    }
+    // push if user has special characters confirmed
+    if (hasSpecialCharacter) {
+      finalRandomPasswordArray.push(specialCharacters);
+    }
+    // returning function with final password array
+    return finalRandomPasswordArray;
+  } else {
+    // start over if they don't select anything
+    alert("Please select at least one criteria for your password");
+    passwordCriteria();
```

Comment:
If you are recursively calling the same function again when no password criteria is chosen you need to ensure that you add the `return` keyword as that will return the value to the parent function in the recursive chain.

```suggestion
    return passwordCriteria();
```

Reply:
This is currently throwing an error when the user fails to select yes for one and then retries again. You need to fix this to improve your grade.